### DISCUSSION TITLE
Is it possible to replace only some files when upgrading a formula?

### DISCUSSION THREAD

**Discussion by cesarcoatl** [Homebrew/discussions#730] [Everyday usage] [answered]
I was wondering if it was possible to keep some folders from a previous version when upgrading a formula.

I have only written versioned formulae for software that persists data inside its installed location, but I am interested in writing a formula that would install or upgrade to the latest version while keeping that persisted data between versions.

There is no build process for the formula, only a simple unzip and install to `libexec`.

Any guidance would be much appreciated.

**Comment by SMillerDev** [comment, marked as answer]
If you make it write to a folder that isn't managed by the formula, for example `prefix/var/somename` your formula will leave the data alone.
